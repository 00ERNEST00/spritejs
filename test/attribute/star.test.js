import StarAttr from '../../src/attribute/star';

describe('attribute set get', () => {
  const subject = {
    onPropertyChange(key, value, oldValue, attributes) { },
  };
  const attr = new StarAttr(subject);

  test('get set', () => {
    expect(attr.d).toBe('');
    attr.innerRadius = 5;
    attr.outerRadius = 15;
    expect(attr.radius).toEqual([5, 15]);
    expect(attr.d).toMatchSnapshot();
    attr.radius = [15, 30];
    expect(attr.d).toMatchSnapshot();
    attr.angles = 3;
    expect(attr.d).toMatchSnapshot();
    attr.offsetAngle = 60;
    expect(attr.d).toMatchSnapshot();
  });
});