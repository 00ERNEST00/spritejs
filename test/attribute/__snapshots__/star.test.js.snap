// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attribute set get get set 1`] = `"M9.18485099360515e-16 -15L2.938926261462366 -4.045084971874737L14.265847744427303 -4.635254915624211L4.755282581475767 1.545084971874737L8.816778784387097 12.135254915624213L3.061616997868383e-16 5L-8.816778784387095 12.135254915624213L-4.755282581475767 1.5450849718747375L-14.265847744427305 -4.635254915624209L-2.9389262614623664 -4.045084971874736Z"`;

exports[`attribute set get get set 2`] = `"M1.83697019872103e-15 -30L8.816778784387097 -12.135254915624213L28.531695488854606 -9.270509831248422L14.265847744427303 4.635254915624211L17.633557568774194 24.270509831248425L9.18485099360515e-16 15L-17.63355756877419 24.270509831248425L-14.265847744427303 4.635254915624213L-28.53169548885461 -9.270509831248418L-8.816778784387099 -12.13525491562421Z"`;

exports[`attribute set get get set 3`] = `"M1.83697019872103e-15 -30L12.990381056766578 -7.5L25.98076211353316 14.999999999999995L9.18485099360515e-16 15L-25.980762113533153 15.00000000000001L-12.990381056766578 -7.500000000000002Z"`;

exports[`attribute set get get set 4`] = `"M25.980762113533157 -15L12.99038105676658 7.499999999999997L8.49830834647197e-15 30L-12.990381056766577 7.500000000000005L-25.98076211353317 -14.999999999999982L-2.7554552980815444e-15 -15Z"`;
